tomviz/ExternalPythonExecutor.cxx

```diff
@@ -0,0 +1,166 @@
+/* This source file is part of the Tomviz project, https://tomviz.org/.
+   It is released under the 3-Clause BSD License, see "LICENSE". */
+
+#include "ExternalPythonExecutor.h"
+#include "DataSource.h"
+#include "EmdFormat.h"
+#include "Operator.h"
+#include "OperatorPython.h"
+#include "Pipeline.h"
+#include "PipelineWorker.h"
+
+namespace tomviz {
+
+ExternalPythonExecutor::ExternalPythonExecutor(Pipeline* pipeline)
+  : ExternalPipelineExecutor(pipeline)
+{
+}
+
+ExternalPythonExecutor::~ExternalPythonExecutor() = default;
+
+Pipeline::Future* ExternalPythonExecutor::execute(vtkDataObject* data,
+                                                  QList<Operator*> operators,
+                                                  int start, int end)
+{
+
+  auto future = ExternalPipelineExecutor::execute(data, operators, start, end);
+
+  // We are now ready to run the pipeline
+  QStringList args = executorArgs(start);
+
+  PipelineSettings settings;
+  auto pythonExecutable = settings.externalPythonExecutablePath();
+
+  // Find the tomviz-pipeline executable
+  auto pythonExecutableFile = QFileInfo(pythonExecutable);
+
+  if (!pythonExecutableFile.exists()) {
+    displayError("External Python Error",
+                 QString("The external python executable doesn't exist: %1\n")
+                   .arg(pythonExecutable));
+
+    return Pipeline::emptyFuture();
+  }
+
+  auto baseDir = pythonExecutableFile.dir();
+  auto tomvizPipelineExecutable =
+    QFileInfo(baseDir.filePath("tomviz-pipeline"));
+  if (!tomvizPipelineExecutable.exists()) {
+    displayError(
+      "External Python Error",
+      QString("Unable to find tomviz-pipeline executable, please ensure "
+              "tomviz package has been installed in python environment."));
```

Comment:
Maybe just add something like "Click the Help button for more details on setting up your Python environment."

Reply:
Done